internal/pkg/symnotify/symnotify.go

```diff
@@ -0,0 +1,107 @@
+// package symnotify provides a file system watcher that notifies events for symlink targets.
+//
+package symnotify
+
+import (
+	"io"
+	"io/ioutil"
+	"github.com/ViaQ/logerr/log"
+	"math"
+	"os"
+	"path/filepath"
+	"time"
+	"github.com/fsnotify/fsnotify"
+)
+
+type Event = fsnotify.Event
+type Op = fsnotify.Op
+
+
+const (
+	Create Op = fsnotify.Create
+	Write     = fsnotify.Write
+	Remove    = fsnotify.Remove
+	Rename    = fsnotify.Rename
+	Chmod     = fsnotify.Chmod
+)
+
+// Watcher is like fsnotify.Watcher but also notifies on changes to symlink targets
+type Watcher struct {
+	watcher *fsnotify.Watcher
+}
+
+
+func NewWatcher() (*Watcher, error) {
+	w, err := fsnotify.NewWatcher()
+	return &Watcher{watcher: w}, err
+}
+
+// Event returns the next event.
+func (w *Watcher) Event() (e Event, err error) {
+	return w.EventTimeout(time.Duration(math.MaxInt64))
+}
+
+// EventTimeout returns the next event or os.ErrDeadlineExceeded if timeout is exceeded.
+func (w *Watcher) EventTimeout(timeout time.Duration) (e Event, err error) {
+	var ok bool
+	select {
+	case e, ok = <-w.watcher.Events:
+	case err, ok = <-w.watcher.Errors:
+	case <-time.After(timeout):
+		return Event{}, os.ErrDeadlineExceeded
+	}
+	switch {
+	case !ok:
+		return Event{}, io.EOF
+	case e.Op == Create:
+		log.V(2).Info("Create Event Detected for file.." ,"e.Name",e.Name)
+		if info, err := os.Lstat(e.Name); err == nil {
+			if isSymlink(info) {
+				_ = w.watcher.Add(e.Name)
+			}
+		}
+	case e.Op == Remove:
+		log.V(2).Info("Remove Event Detected for file.." ,"e.Name",e.Name)
```

Comment:
Should we not actually remove the entry to keep the map from growing?

Reply:
@jcantrill   Removing the path on the issue of Remove event didn't work as per expected results.  Metric counter was not showing increasing bytes. If we don't do  anything, then i found it is working as per expected results.

Reply:
but do we end up with a memory link for a map that grows without bounds?

Reply:
I dont see we have an answer to this questino

Reply:
@alanconway @jcantrill the below is found not yielding expected results ..  
               case e.Op == Remove:
               log.V(2).Info("Remove Event Detected for file.." ,"e.Name",e.Name)
               w.watcher.Remove(e.Name)
               
I don't know the answers to it... why..